apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -310,12 +350,28 @@ defmodule Lexical.RemoteControl.CodeIntelligence.DefinitionTest do
     end
   end
 
-  defp definition(project, code) do
+  defp definition(project, code, referenced_uri \\ nil) do
     with {position, code} <- pop_cursor(code),
          {:ok, document} <- subject_module(project, code),
+         :ok <- index(document, referenced_uri),
          {:ok, %Location{} = location} <-
            RemoteControl.Api.definition(project, document, position) do
       {:ok, location.document.uri, decorate(location.document, location.range)}
     end
   end
+
+  defp index(document, referenced_uri) do
+    {path, text} =
+      if referenced_uri do
+        path = Document.Path.from_uri(referenced_uri)
+        text = File.read!(path)
+        {path, text}
+      else
+        {document.path, Document.to_string(document)}
+      end
+
+    with {:ok, entries} <- Search.Indexer.Source.index(path, text) do
+      Search.Store.replace(entries)
+    end
+  end
```

Comment:
why read directly off the filesystem when we have the document store? Can't you use `open_temporarily`?

Reply:
Ok, changed